sqe-database-access/SearchRepository.cs

```diff
@@ -0,0 +1,197 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Dapper;
+using Microsoft.Extensions.Configuration;
+using SQE.DatabaseAccess.Models;
+
+namespace SQE.DatabaseAccess
+{
+	public interface ISearchRepository
+	{
+		Task<IEnumerable<uint>> SearchEditions(uint userId, string editionName, bool exact);
+
+		Task<List<TextFragmentSearch>> SearchTextFragments(
+				uint                userId
+				, string            textFragmentName
+				, IEnumerable<uint> editionIds
+				, bool              exact);
+
+		Task<List<EditionArtefact>> SearchArtefacts(
+				uint                userId
+				, string            artefactName
+				, IEnumerable<uint> editionIds
+				, bool              exact);
+
+		Task<IEnumerable<SearchImagedObject>> SearchImagedObjects(
+				string imagedObjectName
+				, bool exact);
+	}
+
+	public class SearchRepository : DbConnectionBase
+									, ISearchRepository
+	{
+		public SearchRepository(IConfiguration config) : base(config) { }
+
+		public async Task<IEnumerable<uint>> SearchEditions(
+				uint     userId
+				, string editionName
+				, bool   exact)
+		{
+			using (var conn = OpenConnection())
+			{
+				var sql = @"
+SELECT manuscript_data_owner.edition_id
+FROM manuscript_data
+JOIN manuscript_data_owner USING(manuscript_data_id)
+JOIN edition USING(edition_id)
+JOIN edition_editor ON edition_editor.edition_id = edition.edition_id
+WHERE manuscript_data.name $Match
+    AND (edition.public = 1 OR edition_editor.user_id = @UserId)
+LIMIT 100
+";
+
+				sql = sql.Replace(
+						"$Match"
+						, exact
+								? "= @ManuscriptName"
+								: "LIKE CONCAT('%', @ManuscriptName, '%')");
```

Comment:
Ah, I wondered how 'exact' was implemented (or rather, how inexact was implemented). We should think about ordering results based on the distance of the inexact match from the result. So that an exact match will appear before inexact ones.